publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/batch/EventBasedBatchScanHandler.java

```diff
@@ -0,0 +1,88 @@
+package no.unit.nva.publication.events.handlers.batch;
+
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.time.Clock;
+import no.unit.nva.events.handlers.EventHandler;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.publication.events.bodies.ScanDatabaseRequest;
+import no.unit.nva.publication.events.handlers.PublicationEventsConfig;
+import no.unit.nva.publication.model.ListingResult;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.publication.storage.model.ResourceUpdate;
+import nva.commons.core.JacocoGenerated;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+
+public class EventBasedBatchScanHandler extends EventHandler<ScanDatabaseRequest, Void> {
```

Comment:
Handler that scans the database and updates the row version of the data entries. This will trigger their expansion and persistence in the dedicated S3 folder.